src/mpc-hc/mplayerc.cpp

```diff
@@ -347,207 +349,212 @@ CStringA GetContentType(CString fn, CAtlList<CString>* redir)
+            buffer[0] = '\0';
+            if (redir && (content == _T("audio/x-scpls") || content == _T("audio/scpls")
+                || content == _T("audio/x-mpegurl") || content == _T("audio/mpegurl")
+                || content == _T("video/x-ms-asf") || content == _T("text/plain")
+                || content == _T("application/octet-stream") || content == _T("application/pls+xml"))) {
                 while (body.GetLength() < 64 * 1024) { // should be enough for a playlist...
-                    CStringA str;
-                    str.ReleaseBuffer(s.Receive(str.GetBuffer(256), 256)); // SOCKET_ERROR == -1, also suitable for ReleaseBuffer
-                    if (str.IsEmpty()) {
+                    br = versionFile->Read(buffer, 256);
+                    if (br == 0) {
                         break;
                     }
-                    body += str;
+                    buffer[br] = '\0';
+                    body += buffer;
                 }
             }
+
+            versionFile->Close(); // Close() isn't called by the destructor
+            delete versionFile;
         }
     }
 
-    // Try to guess from the extension if we don't have much info yet
-    if (!fn.IsEmpty() && (ct.IsEmpty() || ct == _T("text/plain"))) {
+    // If content type is empty, plain text or octet-stream (weird server!) GUESS by extension if it exists.....
+    if (!fn.IsEmpty() && (content.IsEmpty() || content == _T("text/plain") || content == _T("application/octet-stream"))) {
         CPath p(fn);
         CString ext = p.GetExtension().MakeLower();
         if (ext == _T(".mpcpl")) {
-            ct = _T("application/x-mpc-playlist");
-        }  else if (ext == _T(".cue")) {
-            ct = _T("application/x-cue-sheet");
-        } else if (ext == _T(".pls")) {
-            ct = _T("audio/x-scpls");
+            content = _T("application/x-mpc-playlist");
+        }
+        else if (ext == _T(".cue")) {
+            content = _T("application/x-cue-sheet");
+        }
+        else if (ext == _T(".pls")) {
+            content = _T("audio/x-scpls");
             parsefile = true;
-        } else if (ext == _T(".m3u") || ext == _T(".m3u8")) {
-            ct = _T("audio/x-mpegurl");
-        } else if (ext == _T(".bdmv")) {
-            ct = _T("application/x-bdmv-playlist");
-        } else if (ext == _T(".asx")) {
-            ct = _T("video/x-ms-asf");
+        }
+        else if (ext == _T(".m3u") || ext == _T(".m3u8")) {
+            content = _T("audio/x-mpegurl");
+        }
+        else if (ext == _T(".bdmv")) {
+            content = _T("application/x-bdmv-playlist");
+        }
+        else if (ext == _T(".asx")) {
+            content = _T("video/x-ms-asf");
             parsefile = true;
-        } else if (ext == _T(".swf")) {
-            ct = _T("application/x-shockwave-flash");
-        } else if (ext == _T(".qtl")) {
-            ct = _T("application/x-quicktimeplayer");
+        }
+        else if (ext == _T(".swf")) {

```

Comment:
@t3chnical1: For example this line, there is an additional newline.

Reply:
I agree. This got by me. I will be more careful in the future.